MapboxNavigation/RouteController.swift

```diff
@@ -112,7 +112,13 @@ extension RouteController: CLLocationManagerDelegate {
         let metersInFrontOfUser = location.speed * RouteControllerDeadReckoningTimeInterval
         let locationInfrontOfUser = location.coordinate.coordinate(at: metersInFrontOfUser, facing: location.course)
         let newLocation = CLLocation(latitude: locationInfrontOfUser.latitude, longitude: locationInfrontOfUser.longitude)
-        return newLocation.isWithin(RouteControllerMaximumDistanceBeforeRecalculating, of: routeProgress.currentLegProgress.currentStep)
+        var radius = RouteControllerMaximumDistanceBeforeRecalculating
+        
+        if 0...RouteControllerMaximumDistanceBeforeRecalculating ~= location.horizontalAccuracy {
+            radius = location.horizontalAccuracy + RouteControllerHorizontalAccuracyAnomaly
```

Comment:
More succinctly, we could set `RouteControllerMaximumDistanceBeforeRecalculating` to 45 meters, then set `radius` to `min(location.horizontalAccuracy, RouteControllerMaximumDistanceBeforeRecalculating) + RouteControllerHorizontalAccuracyAnomaly`.